### PR TITLE
GROK-10717

### PR DESCRIPTION
**Pull request by Aleksashka11**
JS API Tests: DataFrame: test each exposed method and property

### CODE REVIEW

File: dataframe.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  df1.columns.add(DG.Column.fromStrings('countries', ['USA', 'Canada']));
+  df1.columns.add(DG.Column.fromInt32Array('population', Int32Array.from([1, 4])));
+  let df2 = DG.DataFrame.create(2);
+  df2.columns.add(DG.Column.fromStrings('countries', ['France', 'Mexico']));
+  df2.columns.add(DG.Column.fromInt32Array('population', Int32Array.from([2, 3])));
+  let df = DG.DataFrame.create(4);
+  df.columns.add(DG.Column.fromStrings('countries', ['USA', 'Canada', 'France', 'Mexico']));
+  df.columns.add(DG.Column.fromInt32Array('population', Int32Array.from([1, 4, 2, 3])));
+  
+  test('append method', async () => {
```
Comment (skalkin) [inline dataframe.ts:65]: I wonder if async is needed here?

File: dataframe.ts [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    return df1.rows.filter((row) => row.countries === 'USA');
+  });
+
+  test('row list insertAt', async () => {
+    df1.rows.insertAt(2,2);
+    expect(df1.get('countries', 2), '');
+  });
+
+  test('row list match', async () => {
+    return df1.rows.match('countries = USA').highlight();
```
Comment (skalkin) [inline dataframe.ts:239]: Would be nice to check if matching returns proper answers (for instance, you can check the number of matching rows)

**Review by skalkin — APPROVED** [review]
Looks great to me, thanks Oleksandra!

**Comment by onuf** [comment]
@Aleksashka11  Thanks a lot for your contribution. Let's move this file to the `ApiTests` package: https://github.com/datagrok-ai/public/tree/master/packages/ApiTests/src/dataframe

**Pull request merged**